rollup/rollupsyncservice/rollupsyncservice.go

```diff
@@ -0,0 +1,419 @@
+	latestConfirmed, err := s.client.getLatestFinalizedBlockNumber(s.ctx)
+	if err != nil {
+		log.Warn("failed to get latest confirmed block number", "err", err)
+		return
+	}
+
+	log.Trace("Sync service fetchRollupEvents", "latestProcessedBlock", s.latestProcessedBlock, "latestConfirmed", latestConfirmed)
+
+	// query in batches
+	for from := s.latestProcessedBlock + 1; from <= latestConfirmed; from += defaultFetchBlockRange {
+		if s.ctx.Err() != nil {
+			return
+		}
+
+		to := from + defaultFetchBlockRange - 1
+		if to > latestConfirmed {
+			to = latestConfirmed
+		}
+
+		logs, err := s.client.fetchRollupEventsInRange(s.ctx, from, to)
+		if err != nil {
+			log.Error("failed to fetch rollup events in range", "fromBlock", from, "toBlock", to, "err", err)
+			return
+		}
+
+		if err := s.parseAndUpdateRollupEventLogs(logs, to); err != nil {
+			log.Error("failed to parse and update rollup event logs", "err", err)
+			return
+		}
+	}
+}
+
+func (s *RollupSyncService) parseAndUpdateRollupEventLogs(logs []types.Log, lastBlock uint64) error {
+	for _, vLog := range logs {
+		switch vLog.Topics[0] {
+		case s.l1CommitBatchEventSignature:
+			event := L1CommitBatchEvent{}
+			if err := UnpackLog(s.scrollChainABI, event, "CommitBatch", vLog); err != nil {
+				return fmt.Errorf("failed to unpack commit rollup event log, err: %w", err)
+			}
+			batchIndex := vLog.Topics[1].Big().Uint64()
+			chunkRanges, err := s.getChunkRanges(batchIndex, &vLog)
+			if err != nil {
+				return fmt.Errorf("failed to get chunk ranges, err: %w", err)
+			}
+			rawdb.WriteBatchChunkRanges(s.db, batchIndex, chunkRanges)
+
+		case s.l1RevertBatchEventSignature:
+			event := L1RevertBatchEvent{}
+			if err := UnpackLog(s.scrollChainABI, event, "RevertBatch", vLog); err != nil {
+				return fmt.Errorf("failed to unpack revert rollup event log, err: %w", err)
+			}
+			batchIndex := vLog.Topics[1].Big().Uint64()
+			rawdb.DeleteBatchChunkRanges(s.db, batchIndex)
+
+		case s.l1FinalizeBatchEventSignature:
+			event := L1FinalizeBatchEvent{}
+			if err := UnpackLog(s.scrollChainABI, event, "FinalizeBatch", vLog); err != nil {
+				return fmt.Errorf("failed to unpack finalized rollup event log, err: %w", err)
+			}
+			batchIndex := event.BatchIndex.Uint64()
+			batchHash := event.BatchHash
+			stateRoot := event.StateRoot
+			withdrawRoot := event.WithdrawRoot
+
+			parentBatchMeta, chunks, err := s.getLocalInfo(batchIndex)
+			if err != nil {
+				return fmt.Errorf("failed to get local node info, batch index: %v, err: %w", batchIndex, err)
+			}
+
+			if err := s.validateBatch(batchIndex, batchHash, stateRoot, withdrawRoot, parentBatchMeta, chunks); err != nil {
+				return fmt.Errorf("fatal: validateBatch failed: batch index: %v, err: %w", batchIndex, err)
+			}
+			lastChunk := chunks[len(chunks)-1]
+			lastBlock := lastChunk.Blocks[len(lastChunk.Blocks)-1]
+			rawdb.WriteFinalizedL2BlockNumber(s.db, lastBlock.Header.Number.Uint64())
```

Comment:
```suggestion
			endChunk := chunks[len(chunks)-1]
			endBlock := endChunk.Blocks[len(endChunk.Blocks)-1]
			rawdb.WriteFinalizedL2BlockNumber(s.db, endBlock.Header.Number.Uint64())
```

Reply:
renamed in [65629da](https://github.com/scroll-tech/go-ethereum/pull/515/commits/65629daef071314a5271b91e15672cf64dce1454)